### PR TITLE
Moved openForm(),openSavedForm and prefillAndOpenForm() to FormsInteractor

### LINKED ISSUE #35: Move openForm() and prefillAndOpenForm() methods to FormInteractor
`OdkInteractor` interface should only contain `setupODK` and `resetODK` methods as this interface should only be concerned with High level ODK processes while `openForm`, `openSavedForm` and `prefillAndOpenForm` are specific form level processes and should be moved to `FormsInteractor` interface which is responsible for handling form level processes. The prefill function depends on `updateForm` function which also needs to be moved from `ODKInteractor`.

### PR DESCRIPTION
**Pull request by shivenducs1136**
Closes #35 

#### What has been done to verify that this works as intended?
` The sample app I built and tested for the key functions appears to function with all the features that are currently available. ` 

#### Why is this the best possible solution? Were any other approaches considered?
` The best possible solution is to move the methods, and their implementations to formsInteractor & ODKFormsHandler also we have to use formsInteractor object to use the required methods in the ODKFeatureTesterActivity` 

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
` This change would not affect in user perspective `

#### Does this change require updates to documentation?
` Yes, we need to update the readme.md file and move the required methods to appropriate places. `

### CODE REVIEW

File: sample/src/main/java/io/samagra/oce_sample/ODKFeatureTesterActivity.kt [hunk 1]
```diff
@@ -78,6 +80,7 @@ class ODKFeatureTesterActivity : AppCompatActivity(), View.OnClickListener {
                 currentProjectProvider.getCurrentProject().name
                 formsDatabaseInteractor = ODKProvider.getFormsDatabaseInteractor()
                 networkInteractor = ODKProvider.getFormsNetworkInteractor()
+                formsInterctor = ODKProvider.getFormsInteractor()
```
Comment (chinmoy12c) [inline sample/src/main/java/io/samagra/oce_sample/ODKFeatureTesterActivity.kt]: nit, please correct the spelling of `formsInteractor`

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 2]
```diff
@@ -188,4 +197,112 @@ class ODKFormsHandler @Inject constructor(
...
+                    }
+                    is FormStateEvent.OnFormOpenFailed -> if (event.formId == formId) compositeDisposable.clear()
+                    is FormStateEvent.OnFormSaveError -> if (event.formId == formId) compositeDisposable.clear()
+                    else -> {}
+                }
+            })
+            prefillForm(formId, tagValueMap)
+        }
+    }
+    private fun downloadAndOpenForm(formId: String, context: Context) {
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt:282]: Please add a newline here.

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 3]
```diff
@@ -188,4 +197,112 @@ class ODKFormsHandler @Inject constructor(
...
+        formsNetworkInteractor.downloadFormById(formId, object : FileDownloadListener {
+            override fun onComplete(downloadedFile: File) {
+                openFormWithFormId(formId, context)
+            }
+        })
+    }
+    private fun mediaExists(form: Form): Boolean {
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt:290]: Please add a newline here.

**Review by chinmoy12c — CHANGES_REQUESTED** [review]
Thanks @shivenducs1136 the changes look fine :smile: . Just resolve the last nits.

**Review by shivenducs1136 — COMMENTED** [review]
Hey @chinmoy12c sir I have resolved those minor changes 😄, Hope it will work 👍🏻.

**Pull request merged**